### PR TITLE
boot: reseal when updating boot assets

### PR DESCRIPTION
**Pull request by bboozzoo**
This is stacked on top of #9328. The relevant commit is https://github.com/snapcore/snapd/commit/2528fdbc7427fd0d980548d040a46e6cf436ba33

### CODE REVIEW

**Review by pedronis — APPROVED** [review]
+1, not sure the code here is enough to make spread tests happy though for the case were we don't have sealed keys at all

**Review by mvo5 — APPROVED** [review]
LGTM

File: boot/assets.go [hunk 1]
```diff
@@ -570,8 +572,14 @@ func (o *TrustedAssetsUpdateObserver) observeRollback(bl bootloader.Bootloader,
...
 func (o *TrustedAssetsUpdateObserver) BeforeWrite() error {
-	// TODO:UC20:
-	// - reseal with a given state of modeenv
+	if o.modeenv == nil {
+		// modeenv wasn't even loaded yet, meaning none of the trusted
+		// boot assets was updated
+		return nil
+	}
+	if err := resealKeyToModeenv(o.model, o.modeenv); err != nil {
+		return fmt.Errorf("cannot reseal encryption key: %v", err)
```
Comment (pedronis) [inline boot/assets.go:581]: this should just return the error, or will get repetitive messages

File: boot/assets.go [hunk 2]
```diff
@@ -634,8 +642,9 @@ func (o *TrustedAssetsUpdateObserver) Canceled() error {
 		return fmt.Errorf("cannot write modeeenv: %v", err)
 	}
 
-	// TODO:UC20:
-	// - reseal with a given state of modeenv
+	if err := resealKeyToModeenv(o.model, o.modeenv); err != nil {
+		return fmt.Errorf("cannot reseal encryption key after a canceled update: %v", err)
```
Comment (pedronis) [inline boot/assets.go:646]: we do it rarely but for same reasons we might have to do this like:

"while canceling gadget update: %v"

**Review by pedronis — COMMENTED** [review]
small comments about errors

**Pull request merged**